### PR TITLE
Fix configuration restore

### PR DESCRIPTION
**Pull request by gaelj**
Only write 2 profiles instead of 3.

### CODE REVIEW

File: js/backup_restore.js [hunk 1]
```diff
@@ -636,7 +636,7 @@ function configuration_restore(callback) {
     function configuration_upload(configuration, callback) {
         function upload() {
             var activeProfile = null,
-                profilesN = 3;
```
Comment (mikeller) [inline js/backup_restore.js]: I think we need to have this version gated dependent on the firmware version. 3.0.0 is back up to 3 profiles. Or better, we should make the code able to handle the situation where 3 profiles are in a backup, but the current version of the firmware only supports 2.

Comment (gaelj) [inline js/backup_restore.js]: Agreed. The change you suggest should be part of another PR.
This one fixes a crash, I think it should be merged for the time being.

Comment (mikeller) [inline js/backup_restore.js]: Ok. I am not in favour of deploying 'fixes' that break stuff for other users. Will submit a PR with proper version gating as soon as I get to it.

**Comment by borisbstyle** [comment]
Some targets will actually have 3 profiles in 3.0 version

**Comment by mikeller** [comment]
> Some targets will actually have 3 profiles in 3.0 version

Yeah, and there are also situations where a user restores a config with 3 profiles, but the firmware he is using has only 2 profiles, and it would be nice if it did not break for these cases either.

Will look into this.

**Comment by gaelj** [comment]
Again, this fixes a crash. Configuration restore is just broken With current configurator code.

**Comment by mikeller** [comment]
Sorry, I tried it, and restore does not work, but I did not manage to get the configurator to crash.

I think, instead of reducing the number of profiles to 2 for all users, and throwing away data for users with 3 profiles, it is probably a better fix to store 3 profiles for all users by changing 

https://github.com/gaelj/cleanflight-configurator/blob/412558da65cf693955e0750ce983de38bee576df/js/backup_restore.js#L7

to

```
profilesN = 3;
```

This seems to be working fine for firmware versions with only 2 profiles.

**Comment by gaelj** [comment]
Sorry, by crash I meant the configurator hangs and needs a restart. Bottom line, restore is broken.
Setting the profile count to 3 is even better, I assumed BF realeases only supporting 2 profiles wouldn't support it.

**Comment by gaelj** [comment]
Updated, rebased, squased & force-pushed :)
Thanks for your reviews @mikeller btw.
